nodes/analyzer/mesh_select_mk2.py

```diff
@@ -0,0 +1,379 @@
+    bl_label = 'Select mesh elements'
+    bl_icon = 'UV_SYNC_SELECT'
+
+    modes = [
+            ("BySide", "By side", "Select specified side of mesh", 0),
+            ("ByNormal", "By normal direction", "Select faces with normal in specified direction", 1),
+            ("BySphere", "By center and radius", "Select vertices within specified distance from center", 2),
+            ("ByPlane", "By plane", "Select vertices within specified distance from plane defined by point and normal vector", 3),
+            ("ByCylinder", "By cylinder", "Select vertices within specified distance from straight line defined by point and direction vector", 4),
+            ("EdgeDir", "By edge direction", "Select edges that are nearly parallel to specified direction", 5),
+            ("Outside", "Normal pointing outside", "Select faces with normals pointing outside", 6),
+            ("BBox", "By bounding box", "Select vertices within bounding box of specified points", 7)
+        ]
+
+    def update_mode(self, context):
+        self.inputs['Radius'].hide_safe = (self.mode not in ['BySphere', 'ByPlane', 'ByCylinder', 'BBox'])
+        self.inputs['Center'].hide_safe = (self.mode not in ['BySphere', 'ByPlane', 'ByCylinder', 'Outside', 'BBox'])
+        self.inputs['Percent'].hide_safe = (self.mode not in ['BySide', 'ByNormal', 'EdgeDir', 'Outside'])
+        self.inputs['Direction'].hide_safe = (self.mode not in ['BySide', 'ByNormal', 'ByPlane', 'ByCylinder', 'EdgeDir'])
+
+        updateNode(self, context)
+
+    mode: EnumProperty(name="Mode",
+            items=modes,
+            default='ByNormal',
+            update=update_mode)
+
+    include_partial: BoolProperty(name="Include partial selection",
+            description="Include partially selected edges/faces",
+            default=False,
+            update=updateNode)
+
+    percent: FloatProperty(name="Percent",
+            default=1.0,
+            min=0.0, max=100.0,
+            update=updateNode)
+
+    radius: FloatProperty(name="Radius", default=1.0, min=0.0, update=updateNode)
+
+    def draw_buttons(self, context, layout):
+        layout.prop(self, 'mode')
+        if self.mode not in ['ByNormal', 'EdgeDir']:
+            layout.prop(self, 'include_partial')
+    def draw_buttons_ext(self, context, layout):
+        self.draw_buttons(context, layout)
+        layout.prop(self, 'list_match')
+
+    def rclick_menu(self, context, layout):
+        layout.prop_menu_enum(self, "list_match", text="List Match")
+
+    def sv_init(self, context):
+        self.inputs.new('SvVerticesSocket', "Vertices").is_mandatory = True
+        eds = self.inputs.new('SvStringsSocket', "Edges")
+        eds.nesting_level = 3
+        pols = self.inputs.new('SvStringsSocket', "Polygons")
+        pols.nesting_level = 3
+        d = self.inputs.new('SvVerticesSocket', "Direction")
+        d.use_prop = True
+        d.default_property = (0.0, 0.0, 1.0)
+        d.nesting_level = 3
```

Comment:
would be neat to write it this way
```python
self.sv_new_input('SvVerticesSocket', "Direction", {use_prop=True, default_property=(0.0, 0.0, 1.0), nesting_level=3})
```
or
```python
self.sv_new_input('SvVerticesSocket', "Direction", {
    use_prop=True,
    default_property=(0.0, 0.0, 1.0),
    nesting_level=3})
```

Reply:
I didn't know that was possible, thanks!